proposals/0311-task-locals.md

```diff
@@ -95,33 +95,16 @@ We propose to expose the Task's internal ability to "carry metadata with it" via
 
 Task local values may be read from any function running within an asynchronous context. This includes *synchronous* functions which were called from an asynchronous function. Binding a task-local value however must be performed in an asynchronous function.
 
-Declaring a task local value begins with declaring a `TaskLocalKey` that will be used to retrieve it from the task:
+A task-local must be declared as a static stored property, and annotated using the `@TaskLocal` property wrapper. 
 
 ```swift
 enum MyLibrary {
- 
-    @TaskLocal(default: RequestID.empty)
-    static var requestID
-
-}
-
-struct RequestID { 
-  static var empty: RequestID? { nil }
-}
-```
-
-A task-local must be declared as a static stored property, and annotated using the `@TaskLocal` property wrapper. The type of the property should be left to the compiler to infer, based on the default value passed to the property wrapper. It can also be spelled out explicitly, however then the type must be spelled as `TaskLocal<ValueType>.Access`:
-
-```swift
-enum MyLibrary { 
-  
-    @TaskLocal(default: nil)
-    static var requestID: TaskLocal<RequestID?>.Access
-    
+  @TaskLocal
+  static var requestID: RequestID? 
 }
 ```
 
-> 💡 Note: Once property wrappers are allowed on top-level global declarations, it would be possible to declare top-level task locals.
+> 💡 Note: Once property wrappers are allowed on top-level global declarations, it would also be possible to declare top-level task locals.
 
 The important thing about declaring a task-local is that there must be one instance of it. Not doing so is very likely a programmer mistake, will lead to increased and unexpected memory use by multiple instances of the task-local key (the wrapper instances), as well as lead to unexpected lookup behavior, because each instance acts as it's own unique key for the task local storage lookups.
```

Comment:
This is a strange paragraph.  Perhaps it can be reworded to be more informational in tone and less cautionary?  Like: 

> Each task-local declaration represents its own, independent task-local storage.  Reading from one declaration will never observe a value stored using a different declaration, even if the declarations look exactly the same.